TESTS/events/queue/main.cpp

```diff
@@ -283,7 +283,12 @@ int timeleft_events[2];
 void check_time_left(EventQueue *queue, int index, int expected)
 {
     const int event_id = timeleft_events[index];
+#if defined(TARGET_NANO100)
+    /* Enlarge tolerance due to low CPU frequency and no cache */
+    TEST_ASSERT_INT_WITHIN(4, expected, queue->time_left(event_id));
```

Comment:
Can this be based on sys clock rather? What is the value for NANO100 ? We should not have this ifdef target in tests

Reply:
NANO100 runs at 48MHz and has no cache. The tolerance is just empirical. Do you have idea to improve this code?

Reply:
using `SystemCoreClock` and make the tolerance based on its value

mbed_platform/atomic test does it for instance

Reply:
@0xc0170 Thanks for your sample. Changed the tolerance policy: for higher CPU frequency (80MHz or above), tolerance is fixed to 2ms as original; for lower CPU frequency (below 80MHz), tolerance is inversely proportional to CPU frequency.